>From eea72a6f70c7b4cf632c3637e12d4689ce6ff0f5 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Thu, 21 Jul 2011 16:28:06 +0200
Subject: [PATCH 12/25] usb-redir: Call qemu_chr_guest_open/close

To let the chardev now we're ready start receiving data. This is necessary
with the spicevmc chardev to get it registered with the spice-server.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 usb-redir.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/usb-redir.c b/usb-redir.c
index 6932beb..d1aafda 100644
--- a/usb-redir.c
+++ b/usb-redir.c
@@ -840,6 +840,8 @@ static int usbredir_initfn(USBDevice *udev)
     udev->auto_attach = 0;
 
     qemu_chr_add_handlers(dev->cs, &usbredir_chr_handlers, dev);
+    /* Let the other side know we are ready */
+    qemu_chr_guest_open(dev->cs);
 
     return 0;
 }
@@ -861,6 +863,7 @@ static void usbredir_handle_destroy(USBDevice *udev)
 {
     USBRedirDevice *dev = DO_UPCAST(USBRedirDevice, dev, udev);
 
+    qemu_chr_guest_close(dev->cs);
     qemu_chr_close(dev->cs);
     /* Note must be done after qemu_chr_close, as that causes a close event */
     qemu_bh_delete(dev->open_close_bh);
-- 
1.7.5.1

